### ISSUE TITLE
Blockchain data sourcing SPV/Neutrino: connect to p2p network to receive relevant block that contains the scripts we are interested to 

### ISSUE THREAD

**Issue by tiero** [tdex-network/tdex-daemon#413]
At the moment, having the unspents cached in-memory only, means that every update/restart, will take lot of time to re-sync the unspents of the wallet, since we need to make network calls for each address we discover.

The definitive approach is to just use the least dependency from the explorer (ie. GET /block) to fetch the chain of blocks, parse them and cache relevant unspents.

`btwallet`  has a `chain` package that could be adapted to support many node backends such as Electrs explorer or even an Elements node via authenticated JSON RPC.
https://github.com/btcsuite/btcwallet/tree/master/chain

**Comment by tiero** [comment]
Linked discussion https://github.com/tdex-network/tdex-daemon/issues/260

**Comment by tiero** [comment]
https://github.com/btcsuite/btcwallet/blob/master/chain/block_filterer.go#L11

**Comment by tiero** [comment]
To better understand compact block filters https://bitcoin.stackexchange.com/questions/86231/whats-the-distinction-between-bip-157-and-bip-158-are-they-supported-by-bitcoi

**Comment by tiero** [comment]
We should research if neutrino client can be used now with Elements 0.21 (support compact block filters AFAIK) or if needs some modification for the different block strucutre of Liquid

**Comment by louisinger** [comment]
> We should research if neutrino client can be used now with Elements 0.21 (support compact block filters AFAIK) or if needs some modification for the different block structure of Liquid.

the main problem comes from the fact that neutrino is very coupled with [btcd](https://github.com/btcsuite/btcd) (in order to be a full node). here is a list of things that I think we should adapt to liquid (not exhaustive):
- [check proof of work of blocks](https://github.com/louisinger/neutrino/blob/1f534519c336bcefd778f34f05a9c6fd79eaa0ad/blockmanager.go#L2708): this should be adapted to Liquid protocol (check the signature of the block for instance?)
- handle peers protocol messages (via [ServerPeer](https://github.com/louisinger/neutrino/blob/1f534519c336bcefd778f34f05a9c6fd79eaa0ad/neutrino.go#L151) struct): this depends on the `btcd.wire` pkg. Here, the problem is that `wire` messages types are not compliant with Liquid types. 
- [manage a pool of peers](https://github.com/louisinger/neutrino/blob/1f534519c336bcefd778f34f05a9c6fd79eaa0ad/neutrino.go#L101): this is using the `btcd.connmng` package.

in conclusion, the "utxo scanner" part doesn't seem complicated to adapt because the filters use the scripts of the transactions (no need to manage the blinding/unblinding). But it's the "full node" part that is not at all covered by go-elements: we need a way to reproduce `btcd.wire` behavior with elements node.

**Comment by tiero** [comment]
>  But it's the "full node" part that is not at all covered by go-elements

mmm AFAIK, the peer protocol is the same as bitcoin, but I can be wrong here. Assuming we can connect via TCP to a node, is btcd failing to handle the connections? Also, would be fine to hardcode the IP addresses of nodes, if that is porblem (peer discovery)

You can test with those `liquid-testnet.sevenlabs.dev:18886` or `liquid-testnet.blockstream.com:18892`
